react/__tests__/Telemarketing.test.tsx

```diff
@@ -43,7 +43,128 @@ describe('<Telemarketing /> component', () => {
         onInputChange={() => {}}
         onSetSession={() => {}}
       />
-    ).asFragment()
-    expect(component).toMatchSnapshot()
+    )
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('should show attendant email', () => {
+    const email = 'attendant@vtex.com'
+    const { getByText } = render(
+      <Telemarketing
+        attendantEmail={email}
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    const attendantEmail = getByText(email)
+    expect(attendantEmail).toBeTruthy()
+  })
+
+  it('should show login button and form', () => {
+    const { getByText, getByPlaceholderText } = render(
+      <Telemarketing
+        attendantEmail="attendant@vtex.com"
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    const loginAs = getByText('Login as')
+    expect(loginAs).toBeTruthy()
+
+    const emailPlaceholder = getByPlaceholderText('Ex: example@mail.com')
+    expect(emailPlaceholder).toBeTruthy()
+
+    const login = getByText('Login')
+    expect(login).toBeTruthy()
+  })
+
+  it('should show client data', () => {
+    const client: Client = {
+      document: 'my client document',
+      phone: '+9999999999999',
+      name: 'my client name',
+      email: 'client@vtex.com',
+    }
+
+    const { getByText } = render(
+      <Telemarketing
```

Comment:
What do you think to create a method to render the component passing the custom props to it?

Reply:
Suggestion: Could use a default object for the props and use [mergeLeft](https://ramdajs.com/docs/#mergeLeft) or [mergeRight](https://ramdajs.com/docs/#mergeRight) to override the properties that change in specific tests

Reply:
I'm prefering being a little more descritive in the tests instead of using an generic render component function.

Reply:
Don't you think you are repeating information throughout the tests?